File: Easy/896. Monotonic Array/896. Monotonic Array.py
from typing import List


class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        increasing = decreasing = False

        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                increasing = True
            elif nums[i] < nums[i - 1]:
                decreasing = True

            if increasing and decreasing:
                # If we find both increasing and decreasing elements, the list is not monotonic.
                return False

        return True

File: Easy/896. Monotonic Array/896. Monotonic Array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean increasing = false;
        boolean decreasing = false;

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[i - 1]) {
                increasing = true;
            } else if (nums[i] < nums[i - 1]) {
                decreasing = true;
            }

            if (increasing && decreasing) {
                // If we find both increasing and decreasing elements, the array is not monotonic.
                return false;
            }
        }

        return true;
    }
}
